performance_budget.routing.yml

```diff
@@ -0,0 +1,51 @@
+entity.performance_budget.collection:
+  path: 'admin/reports/performance-budget'
+  defaults:
+    _entity_list: 'performance_budget'
+    _title: 'Performance Budget'
+  requirements:
+    _permission: 'administer performance budget'
+
+entity.performance_budget.add_form:
+  path: 'admin/reports/performance-budget/add'
+  defaults:
+    _entity_form: 'performance_budget.add'
+    _title: 'Add Performance Budget'
+  requirements:
+    _entity_create_access: 'performance_budget'
+
+entity.performance_budget.canonical:
+  path: 'admin/reports/performance-budget/{performance_budget}'
+  defaults:
+    _controller: '\Drupal\performance_budget\Controller\PerformanceBudgetController::viewRuns'
+    _title_callback: '\Drupal\performance_budget\Controller\PerformanceBudgetController::title'
+  options:
+    parameters:
+      performance_budget:
+        type: performance_budget
+  requirements:
+    _permission: 'performance_budget.view'
```

Comment:
@pobtastic @bighappyface so I was playing with this a little bit...

Our entities don't currently have access handlers in the annotations. 

From https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Entity%21entity.api.php/group/entity_api/8.2.x: 

> **access:** If your configuration entity has complex permissions, you might need an access control handling, implementing \Drupal\Core\Entity\EntityAccessControlHandlerInterface, but most entities can just use the 'admin_permission' annotation instead. Note that if you are creating your own access control handler, you should override the checkAccess() and checkCreateAccess() methods, not access().

So by lacking, an entity access control handler, all of those `_entity_access` lines are meaningless.  I just tested it and I still have view access on the entity with the following requirements setup: 

```
entity.web_page_archive.canonical:
  path: 'admin/config/development/web-page-archive/{web_page_archive}'
  defaults:
    _controller: '\Drupal\web_page_archive\Controller\WebPageArchiveController::viewRuns'
    _title_callback: '\Drupal\web_page_archive\Controller\WebPageArchiveController::title'
  options:
    parameters:
      web_page_archive:
        type: web_page_archive
  requirements:
    _entity_access: 'web_page_archive.this_does_not_exist'
```

Obviously `web_page_archive.this_does_not_exist` is not defined anywhere, so what it seems to be doing is actually just failing over to this line in the entity annotation: 

```
*   admin_permission = "administer web page archive",
```

Furthermore, it doesn't appear we can leave out the requirements section, so I'd propose just changing everything to use the default admin handler for now.  We don't really have complex use cases here, so I think a single permission handling will suffice. 

So for performance budget all routes, we should have this: 

```
  requirements:
    _permission: 'administer performance budget'
```